capstone/src/main/resources/static/event.js

```diff
@@ -1,36 +1,144 @@
-async function getEvents() {
-  const response = await fetch('get-all-events');
-  const jsonEvents = await response.json();
-  //const eventList = document.getElementById('events-container');
+/**
+ * Retrieves events from server
+ */
+function getEvents() {
+  var userType = sessionStorage.getItem("user-type");
+  if (userType == null) {
+    getUserType(false);
+    if (sessionStorage.getItem("user-type") == null) {
+      return;
+    }
+  }
+  var displaySaveButton = userType == "individual";
+  fetch('get-all-events').then(response => response.json()).then((events) => {
+
+    const eventListElement = document.getElementById('events');
+    eventListElement.innerText = ""; // Clear elements in div
 
-  //jsonEvents.forEach(event => eventList.appendChild(createEvent(event)));
+    events.forEach((event) => {
+      eventListElement.appendChild(createEventElement(event, displaySaveButton));
+    })
+  });
 }
 
-function createEvent(event) {
-  const listEventElement = document.createElement('li');
-  listEventElement.className = 'list-events';
+/**
+ * Format event listing
+ * @param event Event from Get call
+ * @return Formatted event ready to add to document
+ */
+function createEventElement(event, displaySaveButton) {
+  const eventElement = document.createElement('li');
+  eventElement.className = 'event';
+
+  // Click for event detail modal
+  eventElement.addEventListener('click', () => {
+    showEventPage(event);
+  });
+
+  // Name
+  const eventNameElement = document.createElement('p');
+  eventNameElement.innerText = event.eventTitle;
 
+  // Time
+  var date = new Date(event.eventDateTime);
+  const eventTimeElement = document.createElement('p');
+  eventTimeElement.innerText = date.toString().substring(0, 21); // Exclude GMT time zone offset
```

Comment:
Not sure if it is because of the date formatting in datastore but some dates show up as being invalid.

Reply:
Actually it only happens with the test data entries.